public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -46,6 +52,25 @@ interface Props {
   onDataChange: (error: string) => void;
 }
 
+const useSetExpressionAndDataSource = () => {
+  const { setValue } = useFormContext<RuleFormValues>();
+  return (updatedQueries: AlertQuery[]) => {
+    // update data source name and expression if it's been changed in the queries from the reducer when prom or loki query
```

Comment:
this was the only way we found to propagate this values while switching from grafana-managed to cloud type